src/m365/planner/commands/roster/roster-remove.spec.ts

```diff
@@ -0,0 +1,152 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import { telemetry } from '../../../../telemetry';
+import auth from '../../../../Auth';
+import { Cli } from '../../../../cli/Cli';
+import { Logger } from '../../../../cli/Logger';
+import Command, { CommandError } from '../../../../Command';
+import request from '../../../../request';
+import { pid } from '../../../../utils/pid';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+const command: Command = require('./roster-remove');
+
+describe(commands.PLAN_REMOVE, () => {
+  const validRosterId = 'CRp0hFSovEedkXtcX3WnS5gAGgch';
+
+  let log: string[];
+  let logger: Logger;
+  let promptOptions: any;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    auth.service.accessTokens[(command as any).resource] = {
+      accessToken: 'abc',
+      expiresOn: new Date()
+    };
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    promptOptions = undefined;
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.delete,
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+    auth.service.accessTokens = {};
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.ROSTER_REMOVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('prompts before removing the specified roster when confirm option not passed with id', async () => {
+    await command.action(logger, {
+      options: {
+        id: validRosterId
+      }
+    });
+
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified plan when confirm option not passed and prompt not confirmed', async () => {
+    const deleteSpy = sinon.spy(request, 'delete');
+    await command.action(logger, {
+      options: {
+        id: validRosterId
+      }
+    });
+    assert(deleteSpy.notCalled);
+  });
+
+  it('Correctly deletes plan by id', async () => {
```

Comment:
Test titles should not start with a capital letter, same goes for the next test.